chain/actions/src/action_queue.rs

```diff
@@ -48,31 +48,31 @@ lazy_static::lazy_static! {
 #[async_trait]
 pub trait TransactionExecutor {
     /// Executes ticket redemption transaction given a ticket.
-    async fn redeem_ticket(&self, ticket: AcknowledgedTicket, domain_separator: Hash) -> Result<Hash>;
+    async fn redeem_ticket(&self, ticket: &ProvableWinningTicket, domain_separator: &Hash) -> Result<Hash>;
 
     /// Executes channel funding transaction (or channel opening) to the given `destination` and stake.
     /// Channel funding and channel opening are both same transactions.
-    async fn fund_channel(&self, destination: Address, balance: Balance) -> Result<Hash>;
+    async fn fund_channel(&self, destination: &Address, balance: &Balance) -> Result<Hash>;
 
     /// Initiates closure of an outgoing channel.
-    async fn initiate_outgoing_channel_closure(&self, dst: Address) -> Result<Hash>;
+    async fn initiate_outgoing_channel_closure(&self, dst: &Address) -> Result<Hash>;
 
     /// Finalizes closure of an outgoing channel.
-    async fn finalize_outgoing_channel_closure(&self, dst: Address) -> Result<Hash>;
+    async fn finalize_outgoing_channel_closure(&self, dst: &Address) -> Result<Hash>;
 
     /// Closes incoming channel.
-    async fn close_incoming_channel(&self, src: Address) -> Result<Hash>;
+    async fn close_incoming_channel(&self, src: &Address) -> Result<Hash>;
 
     /// Performs withdrawal of a certain amount from an address.
     /// Note that this transaction is typically awaited via polling and is not tracked
     /// by the Indexer.
-    async fn withdraw(&self, recipient: Address, amount: Balance) -> Result<Hash>;
+    async fn withdraw(&self, recipient: &Address, amount: &Balance) -> Result<Hash>;
 
     /// Announces the node on-chain given the `AnnouncementData`
-    async fn announce(&self, data: AnnouncementData) -> Result<Hash>;
+    async fn announce(&self, data: &AnnouncementData) -> Result<Hash>;
 
     /// Registers Safe with the node.
-    async fn register_safe(&self, safe_address: Address) -> Result<Hash>;
+    async fn register_safe(&self, safe_address: &Address) -> Result<Hash>;
```

Comment:
Should not use references for `Address` and `Balance`, they are `Copy`.

Reply:
changed